### DISCUSSION TITLE
Can't make basePath to work

### DISCUSSION THREAD

**Discussion by ivan-kleshnin** [nextauthjs/next-auth#4750] [Help] [answered]
I want to run NextJS + Next-Auth under an `/auth` prefix with a `CredentialsProvider` (just exploring the lib). 
From issues/discussions it follows that `basePath` is a supported feature but I can't make it work for the simplest setup:

**next.config.js**
```
module.exports = {
  basePath: "/auth",
}
```

**pages/_app.js**
```
export default function MyApp({Component, pageProps: {session, ...pageProps}}: AppProps) {
  return <>
    <SessionProvider session={session} basePath="/auth"> -- also tried "/auth/api/auth" and "auth"
      <Component {...pageProps}/>
    </SessionProvider>
  </>
}
```

Now `/auth/api/auth/signin` does render a form but for invalid credentials it redirects to `/api/auth/callback/credentials` which is 404 instead of displaying a flash message above the form.

When I open `http://localhost:3000/auth/api/auth/providers` it shows incorrect values:

```
{ 
  credentials: {
    id: "credentials",
    name: "Credentials",
    type: "credentials",
    signinUrl: "http://localhost:3000/api/auth/signin/credentials", -- no basePath!!!
    callbackUrl: "http://localhost:3000/api/auth/callback/credentials" -- no basePath!!!
  }
}
```

**[...nextauth].js**

Is basic, copy-pasted from the docs:

```tsx
export default NextAuth({
  providers: [
    CredentialsProvider({
      name: "Credentials",

      credentials: {
        username: {label: "Username", type: "text"},
        password: {label: "Password", type: "password"},
      },

      async authorize(credentials, req) {
        ...
      }
    })
  ],
```

❗ If I remove `basePath` from both `next.config` and `SessionProvider` everything behaves correctly. 
So the problem is about `basePath`.

What do I miss? 

---

```
"next": "12.1.6",
"next-auth": "^4.5.0",
```

Demo repo: https://github.com/ivan-kleshnin/next-auth-issue

**Comment by ivan-kleshnin** [comment, marked as answer]
## Update

The above issue can be fixed by defining `NEXTAUTH_URL` like:

```
NEXTAUTH_URL=http://localhost:3000/auth/api/auth
```

It's still unclear why it's necessary to set `basePath` twice but whatever 🤷

**Reply by gyurisc** [reply]
Where do you set this environment variable? In your nextjs .env.local or someplace else?

**Reply by ivan-kleshnin** [reply]
I think it was working as a normal ENV so `.env` or `.env.local` worked. But I'm not sure if it applies today, as Next-Auth has gone through multiple code rewrites since 2022. I don't use this library anymore.
